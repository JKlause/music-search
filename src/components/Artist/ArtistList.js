import React from 'react';
import PropTypes from 'prop-types';
import Artist from './Artist';

export default function ArtistList({ artists }) {

  const artistElements = artists.map(artist => (
    <li key={artist.id}>
      <Artist name={artist.name} id={artist.id}/>
    </li>
  ));

  return (
    <ul>
      {artistElements}
    </ul>
  );
}
ArtistList.propTypes = {
  artists: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  })).isRequired
};
