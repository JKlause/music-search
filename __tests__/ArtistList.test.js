import React from 'react';
import { shallow } from 'enzyme';
import ArtistList from '../src/components/Artist/ArtistList';

describe('ArtistList component', () => {
  it('renders ArtistList', () => {
    const wrapper = shallow(<ArtistList artists={[{ id: '1', name: 'Luke' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
