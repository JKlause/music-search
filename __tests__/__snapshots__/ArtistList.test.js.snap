// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistList component renders ArtistList 1`] = `
<ul>
  <li
    key="1"
  >
    <Artist
      id="1"
      name="Luke"
    />
  </li>
</ul>
`;
